test/restClient.spec.js

```diff
@@ -0,0 +1,228 @@
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const restClient = require('../lib').restClient;
+const types = require('admin-on-rest/lib/rest/types');
+
+const findResult = {
+  total: 1,
+  data: [
+    { id: 1 },
+  ]
+};
+const getResult = { id: 1, title: 'gotten' };
+const updateResult = { id: 1, title: 'updated' };
+const createResult = { id: 1, title: 'created' };
+const removeResult = { id: 1, title: 'deleted' };
+
+let aorClient, fakeClient, fakeService;
+function setupClient() {
+  fakeService = {
+    find: sinon.stub().returns(Promise.resolve(findResult)),
+    get: sinon.stub().returns(Promise.resolve(getResult)),
+    update: sinon.stub().returns(Promise.resolve(updateResult)),
+    create: sinon.stub().returns(Promise.resolve(createResult)),
+    remove: sinon.stub().returns(Promise.resolve(removeResult)),
+  };
+
+  fakeClient = {
+    service: (resource) => fakeService,
+  };
+
+  aorClient = restClient(fakeClient);
+}
+
+
+
+describe('Rest Client', function () {
+
```

Comment:
use the same: 
```
let asyncResult; 
```
for all tests, if not you must define in each describe scope. 
Right now somes are missing.